vsintegration/src/FSharp.Editor/LanguageService/LegacyProjectWorkspaceMap.fs

```diff
@@ -99,85 +89,81 @@ type internal LegacyProjectWorkspaceMap(workspace: VisualStudioWorkspaceImpl,
-            // in Roslyn versions once https://github.com/dotnet/roslyn/pull/26931 is merged. Roslyn will still guarantee that once
-            // there is a project in the workspace with the same path, it'll return the ID of that. So this is sufficient to use
-            // in that case as long as we only use it to call GetProject.
-            let fakeProjectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectDisplayName)
-
-            if isNull (workspace.ProjectTracker.GetProject fakeProjectId) then
-                let hierarchy =
-                    site.ProjectProvider
-                    |> Option.map (fun p -> p :?> IVsHierarchy)
-                    |> Option.toObj
-
-                // Roslyn is expecting site to be an IVsHierarchy.
-                // It just so happens that the object that implements IProvideProjectSite is also
-                // an IVsHierarchy. This assertion is to ensure that the assumption holds true.
-                Debug.Assert(not (isNull hierarchy), "About to CreateProjectContext with a non-hierarchy site")
-
-                let projectContext = 
-                    projectContextFactory.CreateProjectContext(
-                        FSharpConstants.FSharpLanguageName,
-                        projectDisplayName,
-                        projectFileName,
-                        projectGuid,
-                        hierarchy,
-                        Option.toObj site.CompilationBinOutputPath)
+            let hierarchy =
+                site.ProjectProvider
+                |> Option.map (fun p -> p :?> IVsHierarchy)
+                |> Option.toObj
+
+            // Roslyn is expecting site to be an IVsHierarchy.
+            // It just so happens that the object that implements IProvideProjectSite is also
+            // an IVsHierarchy. This assertion is to ensure that the assumption holds true.
+            Debug.Assert(not (isNull hierarchy), "About to CreateProjectContext with a non-hierarchy site")
+
+            let projectContext = 
+                projectContextFactory.CreateProjectContext(
+                    FSharpConstants.FSharpLanguageName,
+                    projectDisplayName,
+                    projectFileName,
+                    projectGuid,
+                    hierarchy,
+                    Option.toObj site.CompilationBinOutputPath)
                 
-                // The real project ID that was actually added. See comments for fakeProjectId why this one is actually good.
-                let realProjectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectDisplayName)
+            // The real project ID that was actually added. See comments for fakeProjectId why this one is actually good.
+            let realProjectId = workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFileName, projectDisplayName)
```

Comment:
I've added Id to IWorkspaceProjectContext. Can you just consume that at this point?

Reply:
If I can get a hold of the dev16 preview 1 package, I can.